src/devices/Mcp23xxx/Mcp23xxx.cs

```diff
@@ -438,19 +435,43 @@ private byte GetMappedAddress(Register register, Port port = Port.PortA, BankSty
             return port == Port.PortA ? address : address += 0x10;
         }
 
-        public void OpenPin(int pinNumber) => throw new NotImplementedException();
-
-        public void OpenPin(int pinNumber, PinMode mode) => SetPinMode(pinNumber, mode);
+        protected override void OpenPin(int pinNumber)
+        {
+            // No-op
+        }
 
-        public void ClosePin(int pinNumber)
+        protected override void ClosePin(int pinNumber)
         {
             // No-op
         }
 
-        public bool IsPinOpen(int pinNumber) => throw new NotImplementedException();
+        protected override PinMode GetPinMode(int pinNumber)
+        {
+            ValidatePin(pinNumber);
+
+            // IsBitSet returns true if bitNumber is flipped on in data.
+            bool IsBitSet(byte data, int bitNumber) => (data & (1 << bitNumber)) != 0;
+
+            if (pinNumber < 8)
+            {
+                return IsBitSet(InternalReadByte(Register.IODIR, Port.PortA), pinNumber)
+                       ? PinMode.Input : PinMode.Output;
+            }
+            else
+            {
+                return IsBitSet(InternalReadByte(Register.IODIR, Port.PortB), pinNumber - 8)
+                       ? PinMode.Input : PinMode.Output;
+            }
+        }
+
+        protected override void AddCallbackForPinValueChangedEvent(int pinNumber, PinEventTypes eventTypes, PinChangeEventHandler callback) => throw new NotImplementedException();
+
+        protected override void RemoveCallbackForPinValueChangedEvent(int pinNumber, PinChangeEventHandler callback) => throw new NotImplementedException();
+
+        protected override int ConvertPinNumberToLogicalNumberingScheme(int pinNumber) => pinNumber;
 
-        public PinMode GetPinMode(int pinNumber) => throw new NotImplementedException();
+        protected override WaitForEventResult WaitForEvent(int pinNumber, PinEventTypes eventTypes, CancellationToken cancellationToken) => throw new NotImplementedException();
 
-        public bool IsPinModeSupported(int pinNumber, PinMode mode) => throw new NotImplementedException();
+        protected override bool IsPinModeSupported(int pinNumber, PinMode mode) => true;
```

Comment:
Just spotted this is wrong, this should be `(mode == PinMode.Output || mode == PinMode.Input)`